pkg/resto/registry.go

```diff
@@ -0,0 +1,157 @@
+package resto
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/registry/client/auth"
+	"github.com/docker/distribution/registry/client/transport"
+	"github.com/docker/docker/api/types"
+	dd "github.com/docker/docker/distribution"
+	"github.com/docker/docker/registry"
+	digest "github.com/opencontainers/go-digest"
+)
+
+type myreference string
+
+func (m myreference) String() string {
+	return string(m)
+}
+
+func (m myreference) Name() string {
+	return string(m)
+}
+
+// MediaTypeConfig is the media type used for configuration files.
+const MediaTypeConfig = "application/vndr.docker.config"
+
+// ConfigManifest is a Manifest type holding arbitrary data.
+type ConfigManifest struct {
+	mediaType string
+	payload   []byte
+}
+
+// References returns the objects this Manifest refers to.
+func (c *ConfigManifest) References() []distribution.Descriptor {
+	return nil
+}
+
+// Payload returns the mediatype and payload of this manifest.
+func (c *ConfigManifest) Payload() (string, []byte, error) {
+	return c.mediaType, c.payload, nil
+}
+
+// NewConfigManifest creates and returns an new ConfigManifest.
+func NewConfigManifest(mediaType string, payload []byte) *ConfigManifest {
+	return &ConfigManifest{mediaType, payload}
+}
+
+func init() {
+	distribution.RegisterManifestSchema(MediaTypeConfig, func(b []byte) (distribution.Manifest, distribution.Descriptor, error) {
+		return &ConfigManifest{
+				mediaType: MediaTypeConfig,
+				payload:   b,
+			},
+			distribution.Descriptor{
+				MediaType: MediaTypeConfig,
+				Size:      int64(len(b)),
+				Digest:    digest.SHA256.FromBytes(b),
+			}, nil
+	})
+}
+
+// NewRepository instantiates a distribution.Repository pointing to the given target, with credentials
+func NewRepository(ctx context.Context, endpoint string, repository string, opts RegistryOptions) (distribution.Repository, error) {
+	named := myreference(repository)
+	authConfig := &types.AuthConfig{
+		Username: opts.Username,
+		Password: opts.Password,
+	}
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	apiendpoint := registry.APIEndpoint{
+		Mirror:    false,
+		URL:       url,
+		Version:   2,
+		TLSConfig: &tls.Config{InsecureSkipVerify: opts.Insecure},
```

Comment:
Do not use `InsecureSkipVerify`. This should never be the default.

Reply:
It's not the default. It's disabled by default, and enabled by a cli argument.